rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+profile = "default" # include rustfmt, clippy
+channel = "nightly-2024-07-15"
```

Comment:
As far as I can see this only gets nightly and I recall that it is not possible to install multi-channel installations through rust-toolchain file. We use stable channel for builds, nightly for coverage and that should stay like that as building with a nightly channel might cause problems: https://github.com/rust-lang/rust/issues/120804 and an instance of this happening around February this year with `ahash`: https://github.com/tkaitchuck/aHash/issues/200 

(btw we also depend on ahash, this could also happen to us if we were to use nightly to build https://github.com/FuelLabs/forc-wallet/blob/152bfb48c347363d32dc573c77315baba8d05e5b/Cargo.lock#L41-L42)


This is also not needed and frustrating for developers managing their binaries through nix or using nix-os.  I am ok to introduce rust toolchain files as long as we are aware of its implications and think about what we get and what we lose. It is not a clear win in mine opinion. But let's discuss :)

Reply:
I am resolving this conversation for now. Let's continue with the rust-toolchain file I feel like you like it :) as long as they are not causing the local builds to be done via nightly I am also fine with it. We can always remove later on if anything in that sense is needed 👌